docs/src/model/md-generate.tsx

```diff
@@ -0,0 +1,115 @@
+import path from "path";
+import glob from "glob";
+import fs from "fs";
+
+export const DOCS_PATH = path.join(process.cwd(), "src/content/");
+export const newLineDividerRegEx = /\r\n|\n/;
+
+export const withIndexURL = (pathname: string): string[] => {
+  const docPath = pathname.split("/");
+  if (docPath[docPath.length - 1] === "index") {
+    return [...docPath.slice(0, docPath.length - 1)];
+  }
+  return docPath;
+};
+
+export const withIndexFile = (docPath: string[], isIndex: boolean): string => {
+  const mdFilePath = path.join(
+    DOCS_PATH,
+    `${docPath.join("/")}${isIndex ? "/index" : ""}.md`
+  );
+  return mdFilePath;
+};
+
+export const withCodeElementWrapper = (content: string) =>
+  `
+   \`\`\` 
+    ${content}   
+    \`\`\`
+  `;
+
+export const getEntriesInfo = (
+  line: string
+): {
+  pathname: string;
+  rowsNumbers: [number, number] | null;
+} => {
+  const rowsNumbers: [number, number] | null = line.includes("{")
+    ? (line
+        .substring(line.indexOf("{"))
+        .replace(/[{}]/g, "")
+        .split("-")
+        .map((lineNumberString) => Number(lineNumberString)) as [
+        number,
+        number
+      ])
+    : null;
+
+  const pathname = (
+    rowsNumbers ? line.substring(0, line.indexOf("{")) : line
+  ).replace("<<< @/", "");
+
+  return {
+    pathname,
+    rowsNumbers,
+  };
+};
+
+export const getContentFromRange = (
+  content: string,
+  rowsNumbers: [number, number]
+) => {
+  const [startLineNumber, endLineNumber] = rowsNumbers;
+
+  return content
+    .split(newLineDividerRegEx)
+    .slice(startLineNumber, endLineNumber)
+    .join("\n");
+};
+
+export const withInsertedCodeFromLinks = (content: string) => {
+  return content
+    .split(newLineDividerRegEx)
+    .map((line: string) => {
+      if (!line.startsWith("<<<")) return line;
+
+      const { pathname, rowsNumbers } = getEntriesInfo(line);
+
+      const fileContent = fs.readFileSync(pathname).toString();
+
+      if (!rowsNumbers) return withCodeElementWrapper(fileContent);
+
+      const contentFromRange = getContentFromRange(fileContent, rowsNumbers);
+
+      return withCodeElementWrapper(contentFromRange);
+    })
+    .join("\n");
+};
+
+export const withoutComments = (content: string) => {
+  return content.replace(/<!--[\s\S]*?-->/gm, "");
+};
+
+export const readMDFileFromPathOrIndex = (pathname: string) => {
+  try {
+    return fs.readFileSync(pathname);
+  } catch (err) {
+    return fs.readFileSync(pathname.replace(".md", "/index.md"));
+  }
+};
+
+export const generateFrontMatterTitleFromContent = (content: string) => {
+  return content.split(newLineDividerRegEx)[0].replace(/[#]*/g, "").trim();
```

Comment:
Do we assume that title can be only in the first line? Did you check this against all files? Are there files without title?

Reply:
Yes, we assume that. They has the same approach. 
1) We can't split Heading in two lines in Markdown like this: 
``` 
# Heading first line
   heading second line
   ```
   Second line is not a Heading in this case. 
   So we can split by new line (\n)
   2) I've checked. There are a few cases without a Heading. So at the site - they just put hardcoded title in this case.